### PR TITLE
Fix CID 1368305: Dereference before null check

### PR DESCRIPTION
**Pull request by jablko**
#1302

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1352/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1244/ for details.

File: plugins/experimental/money_trace/money_trace.cc [hunk 1]
```diff
@@ -120,17 +120,13 @@ mt_check_request_header(TSHttpTxn txnp)
...
-        LOG_DEBUG("found money trace header: %s, length: %d", txn_data->client_request_mt_header, length);
         if (nullptr == (contp = TSContCreate(transaction_handler, nullptr))) {
           LOG_ERROR("failed to create the transaction handler continuation");
-          if (nullptr != txn_data) {
-            TSfree(txn_data->client_request_mt_header);
-            TSfree(txn_data);
-          }
         } else {
+          txn_data                                   = allocTransactionData();
+          txn_data->client_request_mt_header         = TSstrndup(hdr_value, length);
```
Comment (maskit) [inline plugins/experimental/money_trace/money_trace.cc]: Shouldn't we check if `txn_data` is not null?

Comment (jablko) [inline plugins/experimental/money_trace/money_trace.cc]: Hey @maskit! Thanks for reviewing this.

I don't think so ... txn_data is already deferenced in allocTransactionData(), so I assume a null check here would just cause another Coverity defect -- dereference before null check. The current implementation doesn't null check txn_data and this patch doesn't change that.

Comment (maskit) [inline plugins/experimental/money_trace/money_trace.cc]: Well, then I think it should be checked in allocTransactionData() too, but let's do that later if it is really needed.

Comment (jablko) [inline plugins/experimental/money_trace/money_trace.cc]: Yes, I like that. Thanks!

**Pull request merged**